components/org.wso2.carbon.crypto.tenant.hsmstore.mgt/src/main/java/org/wso2/carbon/crypto/tenant/hsmstore/mgt/HSMTenantMgtListener.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.tenant.hsmstore.mgt;
+
+import iaik.pkcs.pkcs11.Session;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.base.api.ServerConfigurationService;
+import org.wso2.carbon.core.util.KeyStoreManager;
+import org.wso2.carbon.crypto.api.CryptoContext;
+import org.wso2.carbon.crypto.api.CryptoException;
+import org.wso2.carbon.crypto.provider.hsm.*;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.objecthandlers.CertificateHandler;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.objecthandlers.KeyHandler;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util.SessionHandler;
+import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
+import org.wso2.carbon.stratos.common.exception.StratosException;
+import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
+
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+
+/**
+ * Implementation of {@link TenantMgtListener} which stores private key and public certificate of the
+ * created tenant in the HSM device.
+ */
+public class HSMTenantMgtListener implements TenantMgtListener {
```

Comment:
Don't we have an abstract class for TenantMgtListener ?

Reply:
TenantMgtListener interface is defined in carbon-multitenancy component. I couldn't find any abstract classes of TenantMgtListener defined in there.